src/YesSql.Abstractions/ILogger.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+
+namespace YesSql
+{
+    public interface ILogger
```

Comment:
Maybe because I wanted a way to log Sql specifically, but thinking about it it seems wrong. What would you suggest? Should we just have an `ILoggerFactory` in the configuration? How could we let a dev configure to only see SQL statements in logs? Or should we just render SQL under a specific level? Like Debug or Information? SQL would only output at Information level for instance. And then we could enable inforamtion on `"YesSql"` or just the class we want to look at?

Reply:
I remember actually. It's because it's not available on all the targets we support. so i needed an abstraction. And then in each application you can plug your own logging to ti.

Reply:
IMHO no need to reinvent the wheel, regarding the SQL we can render it under specific level such as `Debug` as you mentioned above or use sort of colorization if we do so, somethings like what I did in the past with [ColorizedConsoleLoggerProvider](http://hishambinateya.com/log-colorization-using-colorizedconsoleloggerprovider)